### PR TITLE
When building rootless docker image move chown&chmod to build stage

### LINKED ISSUE #13577: Docker rootless image size close to twice as big
Looks like dockerfile command `chown & chmod` creates new layer in size of gitea binary that makes image close to twice as big.
see: https://github.com/go-gitea/gitea/blob/master/Dockerfile.rootless#L54 
and: https://hub.docker.com/layers/gitea/gitea/latest-rootless/images/sha256-b55c30e134a5ce42aaf3cb94402c3ac98294ffbbb82ea1a70e9c669a586b0441?context=explore

Maybe file could be chown'ed and chmod'ed before building docker image?

### PR DESCRIPTION
**Pull request by lafriks**
Have not tested, need feedback

Fixes #13577

@sapk could this work?

### CODE REVIEW

File: Dockerfile.rootless [hunk 1]
```diff
@@ -21,6 +21,8 @@ WORKDIR ${GOPATH}/src/code.gitea.io/gitea
 #Checkout version if set
 RUN if [ -n "${GITEA_VERSION}" ]; then git checkout "${GITEA_VERSION}"; fi \
  && make clean-all build
+ && chmod 755 gitea
+ && chown root:root gitea
```
Comment (silverwind) [inline Dockerfile.rootless]: Probably needs some `\`. Generally I recommend this syntax:

```
RUN cmd && \
  cmd && \
  cmd
```

Comment (lafriks) [reply Dockerfile.rootless]: yes, right, I did just more like a proof of concept PR to get opinions, did even directly from GH web so not tested at all :]
fixed :)

File: Dockerfile.rootless [hunk 2]
```diff
@@ -51,7 +53,6 @@ RUN chown git:git /var/lib/gitea /etc/gitea
 
 COPY docker/rootless /
 COPY --from=build-env /go/src/code.gitea.io/gitea/gitea /usr/local/bin/gitea
```
Comment (techknowlogick) [inline Dockerfile.rootless]: `COPY` supports chown as a flag, it may even perhaps support chmod, but I am unsure.

Comment (lafriks) [reply Dockerfile.rootless]: it does support chmod only chown

Comment (lafriks) [reply Dockerfile.rootless]: moved chown to COPY

**Comment by lafriks** [comment]
Anyway does chmod is really needed. imho golang build will already provide binary with such attributes

**Comment by lafriks** [comment]
Tested building locally and seems to be building with correct attributes and owner

**Review by zeripath — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]

**Comment by techknowlogick** [comment]
🚀

**Comment by sapk** [comment]
@techknowlogick yes that the good solution. I overlook this part as I wanted to be sure that the permission was forced but it wasn't efficient.

**Pull request merged**